packages/eslint-plugin/docs/rules/no-untyped-public-signature.md

```diff
@@ -0,0 +1,49 @@
+# Disallow untyped public methods (no-untyped-public-signature)
+
+public methods are meant to be used by code outside of your class. By typing both the parameters and the return type of public methods they will be more readable and easy to use.
+
+## Rule Details
+
+This rule aims to ensure that only typed public methods are declared in the code.
+
+The following patterns are considered warnings:
+
+```ts
+// untyped parameter
+public foo(param1): void {
+}
+
+// untyped return type
+public foo(param1: string) {
+}
+```
```

Comment:
I would also add two examples here - showing that the param cannot be explicitly `any`, and that the return type cannot be explicitly `any`.

I think that this is an important feature to inform users of.